kyaml/yaml/fns.go

```diff
@@ -424,12 +424,43 @@ func Lookup(path ...string) PathGetter {
 	return PathGetter{Path: path}
 }
 
-// Lookup returns a PathGetter to lookup a field by its path and create it if it doesn't already
+// LookupCreate returns a PathGetter to lookup a field by its path and create it if it doesn't already
 // exist.
 func LookupCreate(kind yaml.Kind, path ...string) PathGetter {
 	return PathGetter{Path: path, Create: kind}
 }
 
+// ConventionalContainerPaths is a list of paths at which containers typically appear in workload APIs.
+// It is intended for use with FirstNodeThatExists.
+var ConventionalContainerPaths = [][]string{
+	// e.g. Deployment, ReplicaSet, DaemonSet, Job, StatefulSet
+	{"spec", "template", "spec", "containers"},
+	// e.g. CronJob
+	{"spec", "jobTemplate", "spec", "template", "spec", "containers"},
+	// e.g. Pod
+	{"spec", "containers"},
+	// e.g. PodTemplate
+	{"template", "spec", "containers"},
+}
+
+// FirstNodeThatExists returns the RNode for the first of the paths that exists in the resource.
+// It returns nil if none of the given paths exist, and returns an error if one is encountered
+// during lookup.
+func FirstNodeThatExists(resource *RNode, paths [][]string) (*RNode, error) {
```

Comment:
Is this a good place for this helper / the variable that goes with it? I initially put it in the framework package, but it struck me that it's kinda just a variation on `Lookup` and `LookupCreate` above (though the name I have isn't similar, to try to be clearer about what it does... `LookupWithFallbackPaths` could be another option?)

Reply:
I think this is a good place for this helper to go; I agree that it is similar to `Lookup` and `LookUpCreate`, so I think `LookupWithFallbackPaths` is a good name for it.

Reply:
A caveat however is that `Lookup` and `LookupCreate` are filters while this is not, so maybe we should either keep the distinction or have this helper return a filter. 

Suggestion: change `PathGetter.Path` to be type `[][]string` and change its filter function to contain the logic here (iterating through all the paths and return the first matching RNode). `Lookup` and `LookupCreate` can return a `PathGetter` with a single path, while `LookupWithFallbackPaths` can return a `PathGetter` with multiple paths. 

Wdyt?

Reply:
A caveat however is that `Lookup` and `LookupCreate` return filters while this does not, so maybe we should either keep the distinction or have this helper return a filter. 

Suggestion: maybe we can have something similar to `PathGetter` that has a `Paths` field of type `[][]string` with a filter function containing the logic here. 

Wdyt?

Reply:
I like `LookupFirstMatch` as a name. 👍  
 
 That's a great point that the other two return filters. Since the new type would only have one field, I'm inclined to use `FilterFunc` to create the filter instead of a whole new type. Pushed that change--LMK what you think.

Reply:
Looks good to me!